src/Modules/reservationPopUp.js

```diff
@@ -0,0 +1,189 @@
+    popupTitle.classList.add('text-3xl', 'font-bold');
+    popupTitle.innerHTML = data.title;
+
+    const commentsTitle = document.createElement('h3');
+    commentsTitle.id = 'reservation-counter';
+
+    const reservations = getReservation(data.key).then((res) => {
+      if (res.error) {
+        commentsTitle.textContent = 'Reservations (0)';
+      } else {
+        commentsTitle.textContent = `Reservations (${res.length})`;
+        // console.log('---++++', res)
+        let htmlToAppend = '';
+        res.forEach((item) => {
+          const html = `<li>${item.date_start} - ${item.date_end} || by <strong> ${item.username}</strong></li>`;
+          htmlToAppend += html;
+        });
+        const reserveList = document.querySelector('.comment-list');
+        reserveList.innerHTML = htmlToAppend;
+      }
+    }).catch((err) => {
+      commentsTitle.textContent = 'Reservations (0)';
+      throw new Error(err);
+    });
+    commentsTitle.textContent = `Reservations (${reservations === undefined ? '0' : reservations?.length})`;
+
+    const commentList = document.createElement('ul');
+    commentList.id = 'comment-list';
+    commentList.classList.add('comment-list');
+
+    if (reservations.length > 0) {
+      reservations.forEach((reservation) => {
+        const commentItem = document.createElement('li');
+        commentItem.innerHTML = `${reservation.date_start} - ${reservation.date_end} || by <strong> ${reservation.username}</strong>`;
+        commentList.appendChild(commentItem);
+      });
+    } else {
+      const noCommentItem = document.createElement('li');
+      noCommentItem.innerHTML = 'No reservations yet';
+      commentList.appendChild(noCommentItem);
+    }
+
+    const formTitle = document.createElement('h3');
+    formTitle.innerHTML = 'Add a reservation';
+
+    const commentInput = document.createElement('input');
+    commentInput.type = 'text';
+    commentInput.classList.add('comment-input');
+    commentInput.placeholder = 'Your name';
+
+    const nameInput = document.createElement('input');
+    nameInput.type = 'date';
+    nameInput.classList.add('name-input', 'custom-h');
+    nameInput.placeholder = 'Start Date';
+
+    const endInput = document.createElement('input');
+    endInput.type = 'date';
+    endInput.classList.add('end-Input');
+    endInput.placeholder = 'End Date';
+
+    const submitBtn = document.createElement('button');
+    submitBtn.type = 'submit';
+    submitBtn.innerHTML = 'Reserve';
+
+    const form = document.createElement('form');
+    form.addEventListener('submit', async (event) => {
+      event.preventDefault();
+      const nameInput = event.target.querySelector('.comment-input').value.trim();
+
+      const startDate = event.target.querySelector('.name-input').value.trim();
+      // const startDateFormat = new Date(startDate);
+
+      const endDate = document.querySelector('.end-Input').value.trim();
+      // const endDateFormat = new Date(endDate);
```

Comment:
- [ Optional ] Please consider removing any piece of commented code from your code base, commented code makes your code base unprofessional and is considered a sign of code that is not clean. I highly recommend you do this.